### PR TITLE
handle server side errors well

### PR DESCRIPTION
**Pull request by boutell**
Handle server side errors well, regardless of nested depth in the schema; bring them to light on the front end; removed status prop that was not well used and replaced it with a data field where used.

You can test by altering a schema field type to throw a server side error in some easily reproduced situation. For instance, I did it by making this change to string fields in schema/index.js.

```
object[field.name] = self.apos.launder.string(data[field.name], field.def);
if (object[field.name].length === 5) {
  throw self.apos.error('invalid', 'This server hates five character strings.');
}
```

### CODE REVIEW

**Comment by boutell** [comment]
The way server side errors are reported in our REST APIs changed a bit, I believe I addressed the places where we were paying attention to that and tested, in particular attachments.

**Comment by boutell** [comment]
I am refactoring a little to merge most of this into the mixin so that the media editor has it too.

**Comment by boutell** [comment]
Supported in the media editor too now. Now it's a CMS.

**Review by stuartromanek — APPROVED** [review]
Approving these Great Enhancements, @localghost443 and I need to work out tickets that complete the user experience (error counts on schema tabs, clearing errors while fixing, more hand holding)

File: modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue [hunk 1]
```diff
@@ -84,10 +91,7 @@ export default {
         fields[item.name].value = {
           data: this.value[item.name]
         };
-        // What is this TODO supposed to be? We have error and value already. -Tom
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue:87]: 😆

**Review by abea — APPROVED** [review]

**Pull request merged**